Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3038,46 +3038,74 @@ void WebPageProxy::flushPendingMouseEventCallbacks()
     m_callbackHandlersAfterProcessingPendingMouseEvents.clear();
 }
 
+#if PLATFORM(IOS_FAMILY)
 void WebPageProxy::dispatchWheelEventWithoutScrolling(const WebWheelEvent& event, CompletionHandler<void(bool)>&& completionHandler)
 {
     sendWithAsyncReply(Messages::WebPage::DispatchWheelEventWithoutScrolling(event), WTFMove(completionHandler));
 }
+#endif
 
 void WebPageProxy::handleWheelEvent(const NativeWebWheelEvent& event)
 {
-    WheelEventHandlingResult handlingResult;
-
-    auto rubberBandableEdges = rubberBandableEdgesRespectingHistorySwipe();
-
-#if ENABLE(ASYNC_SCROLLING) && PLATFORM(COCOA)
-    if (m_scrollingCoordinatorProxy) {
-        handlingResult = m_scrollingCoordinatorProxy->handleWheelEvent(platform(event), rubberBandableEdges);
-        if (!handlingResult.needsMainThreadProcessing()) {
-            if (!handlingResult.wasHandled)
-                wheelEventWasNotHandled(event);
-            return;
-        }
-    }
-#else
-    handlingResult.steps = WheelEventProcessingSteps::MainThreadForScrolling;
-#endif
     if (!hasRunningProcess())
         return;
 
     closeOverlayedViews();
 
+    auto handlingResult = WheelEventHandlingResult { WheelEventProcessingSteps::MainThreadForScrolling, false };
+    auto rubberBandableEdges = rubberBandableEdgesRespectingHistorySwipe();
+
+    if (drawingArea()->shouldSendWheelEventsToEventDispatcher()) {
 #if ENABLE(MOMENTUM_EVENT_DISPATCHER)
-    // FIXME: We should not have to look this up repeatedly, but it can also change occasionally.
-    if (event.momentumPhase() == WebWheelEvent::PhaseBegan && preferences().momentumScrollingAnimatorEnabled())
-        m_scrollingAccelerationCurve = ScrollingAccelerationCurve::fromNativeWheelEvent(event);
+        // FIXME: We should not have to look this up repeatedly, but it can also change occasionally.
```

Comment:
Someone recently told me that there is a way to watch changes to this -- we should consider doing that at some point! This is a whole bunch of work every single scroll.